library/ssl_srv.c

```diff
@@ -2142,6 +2248,9 @@ static void ssl_write_renegotiation_ext( mbedtls_ssl_context *ssl,
         return;
     }
 
+    /* TODO: Add bounds check */
```

Comment:
Unfinished todo

Reply:
Yes, that's something that still needs to be discussed and addressed; note that the bounds check cannot be introduced immediately because the extension writing functions currently don't have return codes.

I left a comment below and in the PR description.